### DISCUSSION TITLE
Access Configurations

### DISCUSSION THREAD

**Discussion by Kelvinchow79** [apache/airflow#27485] [General]
I have inherit a airflow software running in a VM with a account have role admin. However when i go admin/configurations, it says it is hidden by administrator. how do i access the configurations with this admin role? Thanks

**Comment by Taragolis** [comment]
@Kelvinchow79 by default Airflow do not expose configs in the UI.
If you want expose it you need to set in Airflow configuration [[webserver] expose_config](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#expose-config) to `True`

**Reply by Kelvinchow79** [reply]
i have two servers, one for developement and the other for production. the production was hidden while dev was shown. which container should i exec into to execute your command? worker? scheduler? etc.. thanks

**Reply by Taragolis** [reply]
This is not a command this is part of the [configurations of airflow](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#configuration-reference) you need to set it in `airflow.cfg` or by environment variable and restart service.

And you need to set it at least by `airflow webserver` service but general recommendation set same airflow configuration for all services: 

```
Use the same configuration across all the Airflow components. 
While each component does not require all, some configurations need to be same otherwise they would not work as expected. 
A good example for that is [secret_key](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#config-webserver-secret-key) which should be same on the Webserver and Worker to allow Webserver to fetch logs from Worker.
```

And the main thing configs not exposed by default due to security reason because it might show in the UI some sensitive information.
So you should be sure why you are you want to expose config in the production and their consequences.

**Reply by Kelvinchow79** [reply]
Oh i see, i just want to see the config, there are complaints from users that dag failure notification was not send to personal email, only to shared teams chat. so i need to see if i could help. By the way, airflow.cfg is in airflow webserver container? Thanks

**Reply by Taragolis** [reply]
> By the way, airflow.cfg is in airflow webserver container? Thanks

Yes it should be in `AIRFLOW_HOME` directory and if image do not contain this file than it would be created (with default values) during container start.

BTW there is nice [HowTo: Setting Configuration Options](https://airflow.apache.org/docs/apache-airflow/stable/howto/set-config.html#setting-configuration-options) exists which might help you.

**Reply by potiuk** [reply]
Generally @Kelvinchow79  - I think every Airflow deployment is different. You need to  really understand how it's done and read airflow docs in detail - especially about setting configurations, deploying airflow etc.

I suggest to start with those pages: https://airflow.apache.org/docs/apache-airflow/stable/installation/index.html  - this contains, a lot of the explanation on how you can install airflow, how you can deploy it and what are the knowledge and skills you need to posses in order to effectively manage such installation - all the topics mentioned there are described in Airflow docs and I suggest to start your journey from there.

Here - people are trying to help in their free time, and you might get some answers (most will be anyhow redirecting you to the docs so it's sometimes more efficient to look there before asking here. I think it's great for cases where things are about "general" and "replicable" problems where your discussion is public and others looking at it might also benefit from resolving those problems  - and many people here are happy to help with those.

But if you get into a corner of your specific installation, which is something that is not "airflow" in general but maybe "how my airflow is installed" where you have no idea, I suggest to consider that your company (which put you in this situation) will invest some of their money into some paid consultancy so that you can learn for your specific case.  There are some new initiatives popping up that makes it easier for such cases to be handled - for example I can recommend (disclaimer - I am part of the initiative) ringerhq.com - where you can get private paid 30 mins consultancy with experts .

**Reply by Kelvinchow79** [reply]
@Taragolis , thanks for the help and webpage suggestion, i will take a look.

@potiuk, thanks for the documentation and insights, i will help myself with the documentations first, and if really have serious problems may ask for consultation help.
